### ISSUE TITLE
Windows on ARM, Support?

### ISSUE THREAD

**Issue by sidd-kishan** [msys2/Cygwin#53]
Please let us know when can we have an ARM64 version for Windows on ARM OS. We can help you test We have Windows on Rasberry Pi setup. Please pursue it we at Windows on Rasberry Pi community will be glad to extend support in testing your drivers and tools for ARM64.

**Comment by jeremyd2019** [comment]
For msys2-runtime, not so much.  This is a fork of Cygwin, and Cygwin is not really ARM ready.

That being said, I have been working on getting a Clang-based ARM64 toolchain and building some [MINGW-packages](https://github.com/msys2/MINGW-packages/) for ARM64.  https://packages.msys2.org/updates?repo=clangarm64

**Comment by jeremyd2019** [comment]
msys2-runtime does run reasonably well under emulation, both i686 and x86_64.  i686 msys2 is no longer officially supported though, but I have also been personally maintaining updated packages for it at https://github.com/jeremyd2019/msys2-build32/releases/tag/repo

**Comment by jeremyd2019** [comment]
Also, #44 (which was upstreamed to cygwin) and #47 improve the behavior of msys2-runtime on ARM64 under emulation.

**Comment by jeremyd2019** [comment]
It might interest you to know also that all of the ARM64 packages thus far have been built on a Raspberry Pi 3B+ running Windows 10 (currently 20H2), so I definitely appreciate the efforts of the Windows on Raspberry Pi community on providing this option.

**Comment by Biswa96** [comment]
@sidd-kishan Convince GitHub to provide a Win ARM64 public shared runner.

**Comment by jeremyd2019** [comment]
actions/virtual-environments#768

**Comment by jeremyd2019** [comment]
There is a pull request to add Windows ARM64 support to the self-hosted runner, if there was a way to 'secure' it how about an impromptu cluster of volunteer Raspberry Pis? 🐢💨 

More seriously though, I have had success running Windows Insider Preview inside kvm/qemu on an AWS bare-metal Graviton2 instance.  The issue with that is that the smallest bare-metal instance is 64 cores and 128GB ram, and prohibitively expensive to run continuously.  Could probably arrange to run 16 to 32 runners in VMs on hardware like that though.

**Comment by Biswa96** [comment]
Or cross compile every clangarm64 packages 🤔

**Comment by jeremyd2019** [comment]
> msys2-runtime does run reasonably well under emulation, both i686 and x86_64. i686 msys2 is no longer officially supported though, but I have also been personally maintaining updated packages for it at https://github.com/jeremyd2019/msys2-build32/releases/tag/repo

I wrote up some quick instructions on how to get a "fresh" i686 msys2 install updated with my updated packages: https://github.com/jeremyd2019/msys2-build32/wiki/Fresh-install

In case those on Raspberry Pi are running the officially released versions that only support i686 emulation.

**Comment by Biswa96** [comment]
Haven't you tried the x64 msys2?

**Comment by jeremyd2019** [comment]
yes.  but x86_64 emulation only works on insider previews.  the officially released Windows 10 versions only support i686.

**Comment by jeremyd2019** [comment]
See https://github.com/msys2/MSYS2-packages/discussions/2889, that seems to have gotten more traction with details on what would be required to port cygwin/msys2-runtime to ARM64.

**Comment by crueter** [comment]
> [@sidd-kishan](https://github.com/sidd-kishan) Convince GitHub to provide a Win ARM64 public shared runner.

Well, it's here!

**Comment by sidd-kishan** [comment]
> > [@sidd-kishan](https://github.com/sidd-kishan) Convince GitHub to provide a Win ARM64 public shared runner.
> 
> Well, it's here!

You mean https://github.com/actions/partner-runner-images

**Comment by crueter** [comment]
Indeed. 

I also have a WoA laptop and an Asahi install, so I am capable of assisting where need be both with the MinGW cross aspect and the Cygwin aspect.

**Comment by dscho** [comment]
I'm not actually sure that you gentle people realize the scope, the enormous size of the task you're asking. Here are a couple of pointers to get you started to just assess the enormity of the project:

- https://github.com/Windows-on-ARM-Experiments has some experimental patches and workflows to get a native Windows/ARM64 Cygwin Bash working.
- https://inbox.sourceware.org/cygwin-patches/?q=aarch64 gives you a coarse impression of the amount of work still necessary
